src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_MultipleSpecialTypeParam_StringReturn.generated.txt

```diff
@@ -175,5 +188,1415 @@ namespace Microsoft.AspNetCore.Http.Generated
                 return httpContext.Response.WriteAsJsonAsync(obj);
             }
         }
+
+        private static async ValueTask<(bool, T?)> TryResolveBody<T>(HttpContext httpContext, bool allowEmpty)
+        {
+            var feature = httpContext.Features.Get<Microsoft.AspNetCore.Http.Features.IHttpRequestBodyDetectionFeature>();
+
+            if (feature?.CanHaveBody == true)
+            {
+                if (!httpContext.Request.HasJsonContentType())
+                {
+                    httpContext.Response.StatusCode = StatusCodes.Status415UnsupportedMediaType;
+                    return (false, default);
+                }
+                try
+                {
+                    var bodyValue = await httpContext.Request.ReadFromJsonAsync<T>();
+                    if (!allowEmpty && bodyValue == null)
+                    {
+                        httpContext.Response.StatusCode = StatusCodes.Status400BadRequest;
+                        return (false, bodyValue);
+                    }
+                    return (true, bodyValue);
+                }
+                catch (IOException)
+                {
+                    return (false, default);
+                }
+                catch (System.Text.Json.JsonException)
+                {
+                    httpContext.Response.StatusCode = StatusCodes.Status400BadRequest;
+                    return (false, default);
+                }
+            }
+            return (false, default);
+        }
     }
-}
+
+    %GENERATEDCODEATTRIBUTE%
+    file class EndpointFilterInvocationContext<T0> : EndpointFilterInvocationContext, IList<object?>
```

Comment:
A couple comments on this:

1. Do we always need to generate all these types, even when they aren't used? Can we just generate the ones that are actually needed?
2. Should we instead move these types into the library instead of generating them in all the apps?

Reply:
> Do we always need to generate all these types, even when they aren't used? Can we just generate the ones that are actually needed?

In https://github.com/dotnet/aspnetcore/issues/46640, I posited that we could probably add some smartness to the generator to only generate the overloads that we need based on what we encounter.

> Should we instead move these types into ASP.NET instead of generating them in all the apps?

They already exist in the framework but are internal. The issue above highlights an approach where we make them public in the framework. This would mean that we can remove the implementation here.

We discussed that this is likely the best approach and I'll do the API review + code change for that separately.

Reply:
Here's the API issue for us to discuss tomorrow: https://github.com/dotnet/aspnetcore/issues/46676